sound/pci/hda/hda_codec.c

```diff
@@ -2936,6 +2936,10 @@ static int hda_codec_runtime_suspend(struct device *dev)
 	struct hda_codec *codec = dev_to_hda_codec(dev);
 	unsigned int state;
 
+	/* Nothing to do if card registration fails and the component driver never probes */
+	if (!codec->card)
+		return 0;
+
```

Comment:
maybe we should add more details, e.g. in the commit message,  on where exactly the dereference happens. the explanation 'while trying to power down the codec widgets' is a bit short and may trigger questions upstream.

Reply:
@plbossart how about now? It is particularly the codec->wcaps that is not allocated memory

Reply:
@plbossart should I send this upstream since Kai is out for the next couple of weeks?

Reply:
Sounds good to me.